handlers/src/unit-test/java/uk/gov/moj/sdt/handlers/WsCreateBulkRequestHandlerTest.java

```diff
@@ -0,0 +1,151 @@
+        wsCreateBulkRequestHandler.setBulkSubmissionValidator(mockBulkSubmissionValidator);
+        wsCreateBulkRequestHandler.setTransformer(mockTransformer);
+        wsCreateBulkRequestHandler.setConcurrencyMap(mockConcurrencyMap);
+
+        startingBulkSubmitCount = getMetricsBulkSubmitCount();
+        startingActiveBulkCustomerCount = SdtMetricsMBean.getMetrics().getActiveBulkCustomers();
+
+        when(mockBulkRequestType.getHeader()).thenReturn(mockHeader);
+        when(mockHeader.getSdtCustomerId()).thenReturn(1L);
+        when(mockHeader.getCustomerReference()).thenReturn("1");
+    }
+
+    @Test
+    public void testSubmitBulk() {
+        BulkResponseType result;
+
+        when(mockTransformer.transformJaxbToDomain(mockBulkRequestType)).thenReturn(mockBulkSubmission);
+        when(mockTransformer.transformDomainToJaxb(mockBulkSubmission)).thenReturn(mockBulkResponseType);
+
+        InFlightMessage inFlightMessage = new InFlightMessage();
+        inFlightMessage.setCompetingThreads(new HashMap<>());
+        when(mockConcurrencyMap.get(anyString())).thenReturn(inFlightMessage);
+
+        try (MockedStatic<VisitableTreeWalker> mockStaticVisitableTreeWalker = Mockito.mockStatic(VisitableTreeWalker.class)) {
+            mockStaticVisitableTreeWalker.when(() -> VisitableTreeWalker.walk(mockBulkSubmission, "Validator"))
+                .then((Answer<Void>) invocation -> null);
+
+            result = wsCreateBulkRequestHandler.submitBulk(mockBulkRequestType);
+
+            verifyStatic(VisitableTreeWalker.class);
+            VisitableTreeWalker.walk(mockBulkSubmission, "Validator");
+
+            verify(mockTransformer).transformJaxbToDomain(mockBulkRequestType);
+            verify(mockBulkSubmissionValidator).checkIndividualRequests(mockBulkSubmission);
+            verify(mockBulkSubmissionService).saveBulkSubmission(mockBulkSubmission);
+            verify(mockTransformer).transformDomainToJaxb(mockBulkSubmission);
+        }
+
+        testMetrics();
+        Assertions.assertEquals(mockBulkResponseType, result);
+    }
+
+    @Test
+    public void testSubmitBulkThrowsAbstractBusinessException() throws Exception {
+        CustomerNotFoundException exception = new CustomerNotFoundException("MOCK_CODE", "MOCK_DESCRIPTION");
+
+        when(mockTransformer.transformJaxbToDomain(mockBulkRequestType))
+            .thenThrow(exception);
+
+        wsCreateBulkRequestHandler.submitBulk(mockBulkRequestType);
+
+        verifyPrivate(wsCreateBulkRequestHandler)
+            .invoke(
+                "handleBusinessException",
+                any(CustomerNotFoundException.class),
+                any(BulkResponseType.class)
+            );
+
+        testMetrics();
+    }
+
+    private static long getMetricsBulkSubmitCount() {
+        return (long) ReflectionTestUtils.invokeGetterMethod(SdtMetricsMBean.getMetrics(), "getBulkSubmitCount");
```

Comment:
instead of this, try to call reset() for each test method. so you don't need to call private method using reflection